File: API/Database/DbModels/Horse.cs
using System.ComponentModel.DataAnnotations;

namespace API.Database.DbModels;

public class Horse
{
    [Key]
    public int Id { get; set; }
    public string Name { get; set; }
    public string Number { get; set; }
    public bool Male { get; set; }
    public int MotherId { get; set; }
    public int FatherId { get; set; }
    public List<Result> Results { get; set; }
    public List<Competition> Competitions { get; set; }
    public List<Image> Images { get; set; }
}

File: client/src/app/modules/administration/models/Horse.ts
import { Rider } from "./Rider";

export class Horse {
    public static clone(h: Horse): Horse {
        return new Horse(h.id, h.name, h.number, h.dateOfBirth, h.male, h.motherId, h.fatherId)
    }

    constructor(
        public id: number = 0,
        public name: string = "",
        public number: string = "",
        public dateOfBirth: string = "",
        public male: boolean = true,
        public motherId: number = 0,
        public fatherId: number = 0,

    ) {
    }
}